app/classes/Transvision/Search.php

```diff
@@ -59,6 +59,26 @@ class Search
     protected $repository;
 
     /**
+     * The type of search we will perform
+     * @var string
+     */
+    protected $search_type;
+
+    /**
+     * The different types of search we can perform
+     * @var array
+     */
+    protected $search_types = ['strings', 'entities', 'strings_entities'];
+
+    /**
+     * The different types of search we can perform
+     * @var array
+     */
+    protected $form_search_options = [
+        'case_sensitive', 'whole_word', 'perfect_match', 't2t', 'repo', 'search_type',
```

Comment:
I would also order both these arrays alphabetically, just to make them easier to read.